integrations/apify/src/client.ts

```diff
@@ -0,0 +1,435 @@
+        data: {
+          runId: run.id,
+          datasetId: run.defaultDatasetId,
+          itemsCount: allItems.length,
+          filesCreated,
+          syncTargetPath: targetPath,
+        },
+      };
+    } catch (error) {
+      this.logger.forBot().error('Error getting run information:', error);
+      
+      if (error instanceof ApifyApiError) {
+        const { message, type, statusCode, clientMethod } = error;
+        this.logger.forBot().error('Apify API Error:', { message, type, statusCode, clientMethod });
+        
+        return {
+          success: false,
+          message: `Apify API Error: ${message}`,
+          data: {
+            error: message,
+            type,
+            statusCode,
+            clientMethod,
+          },
+        };
+      }
+
+      return {
+        success: false,
+        message: error instanceof Error ? error.message : 'Error in getAndSyncRunResults',
+        data: {
+          error: error instanceof Error ? error.message : 'Error in getAndSyncRunResults',
+        },
+      };
+    }
+  }
+
+  private isLastPage(offset: number, limit: number, total: number): boolean {
+    return offset + limit >= total;
+  }
+
+  private getNextOffset(offset: number, limit: number): number {
+    return offset + limit;
+  }
+
+  private async fetchAllDatasetItems(dataset: any) {
+    let allItems: {
+      markdown?: string;
+      html?: string;
+      text?: string;
+      url?: string;
+      metadata?: { url?: string };
+    }[] = [];
+    let offset = 0;
+    const limit = 1000; 
+
+    while (true) {
+      const { items, total } = await dataset.listItems({ limit, offset });
+      
+      this.logger.forBot().info(`Fetched ${items.length} items (offset: ${offset}, total: ${total})`);
+      
+      allItems.push(...items);
+      
+      // If there are no more items to fetch, exit the loading
+      if (this.isLastPage(offset, limit, total)) {
+        break;
+      }
+      
+      offset = this.getNextOffset(offset, limit);
+    }
+
+    return allItems;
+  }
+
+  private async syncContentToBotpress(items: {
+    markdown?: string;
+    html?: string;
+    text?: string;
+    url?: string;
+    metadata?: { url?: string };
+  }[], targetPath: string | undefined, kbId: string): Promise<number> {
+    this.logger.forBot().info(`[FILE SYNC] Starting sync for ${items.length} items${targetPath ? ` to path: ${targetPath}` : ''}`);
+    let filesCreated = 0;
+
+    for (let i = 0; i < items.length; i++) {
+      const item = items[i];
+      
+      this.logger.forBot().info(`[FILE SYNC] Processing item ${i + 1}/${items.length}:`, {
+        hasMarkdown: !!item?.markdown,
+        hasHtml: !!item?.html,
+        hasText: !!item?.text,
+        url: item?.url || item?.metadata?.url,
+        itemKeys: item ? Object.keys(item) : 'undefined'
+      });
+      
+      if (!item) {
+        this.logger.forBot().warn(`[FILE SYNC] Skipping item ${i}: Item is undefined`);
+        continue;
+      }
```

Comment:
question: what's going on here, do we need the () ? If we can remove them, please remove them.